src/main/scala/ru/butik/butifactory/ApkService.scala

```diff
@@ -17,6 +18,12 @@ class ApkService(datastore: Datastore, storageBackend: ArtifactStorageBackend, f
     val version = apk.apkFile.getApkMeta.getVersionName
     val versionCode = apk.apkFile.getApkMeta.getVersionCode
 
+    val buffer = new Array[Byte](8192)
+    val md5Instanse = MessageDigest.getInstance("MD5")
+    val dis = new DigestInputStream(new FileInputStream(apk.file), md5Instanse)
+    try { while (dis.read(buffer) != -1) { } } finally { dis.close() }
```

Comment:
closable resources can throw exception while closing. It will shadow exception in try block.
You should use try-catch in finally block or smt like `better-files` https://github.com/pathikrit/better-files

Reply:
done